### ISSUE TITLE
Cop idea: redundant if-else statement

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#11945]
## Is your feature request related to a problem? Please describe.

I have the code samples:

a)

```ruby
my_field =
  if condition?
    my_field
  else
    Time.now.utc
  end

my_field
```

b)

```ruby
my_field = condition? ? my_field : Time.now.utc

my_field
```

c)

```ruby
my_field =
  if condition?
    Time.now.utc
  else
    my_field
  end

my_field
```

d)

```ruby
my_field = condition? ? Time.now.utc : my_field

my_field
```

Rubocop has not detected any offenses.

```bash
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop example.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## Describe the solution you'd like

I expected to have the following code samples:

a)

```ruby
unless condition?
  my_field = Time.now.utc
end

my_field
```

b)

```ruby
if condition?
  my_field = Time.now.utc
end

my_field
```

**Comment by l3x4** [comment]
Maybe one-liner?
```ruby
my_field = Time.now.utc if(unless) condition?
```

If the last `my_field` in your examples is an implicit return value, it can be also done like this:
```ruby
def some_meth(my_field)
  return Time.now.utc if(unless) condition

  my_field
end
```
